src/lib.rs

```diff
@@ -1,16 +1,21 @@
+use std::ffi::NulError;
 use std::os::raw::c_int;
 use std::{error, fmt};
 
 #[derive(PartialEq, Eq, Debug)]
 pub enum Error {
     UnsupportedPlatform,
+    Promises(NulError),
+    Execpromises(NulError),
```

Comment:
we no longer unwrap and panic on `CString::new` failures, but let me know if that was intentional

Reply:
Nope, mapping the failure is better